File: src/main/java/com/example/ex3_2_back/entity/Genre.java
package com.example.ex3_2_back.entity;


import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "genre")
@Builder
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Genre {
    @Id
    @Column(name = "genre_id")
    @GeneratedValue(strategy = GenerationType.AUTO)
    Integer id;
    @OneToOne
    @JoinColumn(name = "genre_hub_id")
    GenreHub genreHub;

    @OneToOne
    @JoinColumn(name = "movie_id")
    Movie movie;
}

File: src/main/kotlin/com/example/ex3_2_back/entity/Genre.kt
package com.example.ex3_2_back.entity

import jakarta.persistence.*

@Entity
@Table(name = "genre")
class Genre(
    @Id
    @Column(name = "genre_id")
    var id: Int = 0,
    @OneToOne
    @JoinColumn(name = "genre_hub_id")
    var genreHub: GenreHub = GenreHub(),
    @OneToOne
    @JoinColumn(name = "movie_id")
    var movie: Movie = Movie(),
)